tests/testthat/test-categories.R

```diff
@@ -318,6 +318,44 @@ with_test_authentication({
             expect_equivalent(as.array(crtabs(~ v4e, data=ds)),
                 array(c(10, 10), dim=2L, dimnames=list(v4=c("B", "D"))))
         })
+
+        test_that("Can changeCategoryID", {
+            ds$v4f <- df$v4
+            expect_identical(names(categories(ds$v4f)),
+                c("B", "C", "No Data"))
+            expect_equal(ids(categories(ds$v4f)),
+                c(1, 2, -1))
+            ds$v4f <- changeCategoryID(ds$v4f, 2, 6)
+            expect_identical(names(categories(ds$v4f)),
+                c("B", "C", "No Data"))
+            expect_equal(ids(categories(ds$v4f)),
+                c(1, 6, -1))
+            expect_error(ds$v4f <- changeCategoryID(ds$v4f, 2, 7),
+                "No category with id 2")
+            expect_error(ds$v4f <- changeCategoryID(ds$v4f, 1, 6),
+                         "Id 6 is already a category, please provide a new category id.")
+            expect_error(ds$v4f <- changeCategoryID(ds$v4f, "not a numeric", 7),
+                         "The 'from' argument is not a numeric, please providee only the id number you want to change from")
+            expect_error(ds$v4f <- changeCategoryID(ds$v4f, 6, "not a numeric"),
+                         "The 'to' argument is not a numeric, please providee only the id number you want to change to")
+            ds$v3f <- df$v3
+            expect_error(ds$v3f <- changeCategoryID(ds$v3f, 2, 7),
+                         "The variable v3f doesn't have categories.")
+        })
+
+        test_that("Can changeCategoryID for array variables", {
+            ds_apidocs <- newDatasetFromFixture("apidocs")
+            ds_apidocs$petloca <- ds_apidocs$petloc
```

Comment:
I mean you don't need a copy of `petloc` to operate on because you've created this dataset just for this test. The example above where I add a variable to an existing dataset to work with is atypical.

(Also, it turns out that `ds_apidocs$petloca <- ds_apidocs$petloc` is just a renaming.)